### DISCUSSION TITLE
Create table with primary keys, from a query

### DISCUSSION THREAD

**Discussion by sami-ka** [duckdb/duckdb#6626] [Q&A] [answered]
Hi, 

I would like to create a table from a query that runs on a set of parquet files in python. 
I try to indicate what are the columns that are the primary keys of this table, as the end goal is to have faster access/queries to it.

I tried the following syntax with some small changes here and there but unfortunately I do not manage to make it work. 

```
request_table_creation = '''CREATE TABLE T(PRIMARY KEY (date, id)) AS 
(SELECT *
FROM '/tmp/tmp123/br_2*.parquet')'''

table_T = duckdb.query(request_table_creation)
```

I get the following syntax error :

```
duckdb.ParserException: Parser Error: syntax error at or near "AS"
LINE 1: ...TABLE T(PRIMARY KEY (date, id)) AS SELECT *
```

Is there a way to make it work or is duckdb not supposed to handle that type of request?

**Comment by Mytherin** [comment, marked as answer]
That does not work right now no. You can create the table first and then load data into it using an `INSERT INTO` statement from a query instead.

**Reply by sami-ka** [reply]
Thanks a lot for your quick answer @Mytherin , perfectly clear! 
And also thanks for the tremendous work on this package 👍
